src/pip/_internal/operations/install/wheel.py

```diff
@@ -573,6 +575,17 @@ def is_entrypoint_wrapper(name):
     shutil.move(temp_installer, installer)
     generated.append(installer)
 
+    # Record the PEP 610 direct URL reference
+    if direct_url is not None:
+        direct_url_path = os.path.join(dest_info_dir, DIRECT_URL_METADATA_NAME)
+        temp_direct_url_path = os.path.join(
+            dest_info_dir, DIRECT_URL_METADATA_NAME + ".pip"
+        )
+        with open(temp_direct_url_path, 'wb') as temp_direct_url_file:
+            temp_direct_url_file.write(direct_url.to_json().encode("utf-8"))
+        shutil.move(temp_direct_url_path, direct_url_path)
```

Comment:
Issue created: https://github.com/pypa/pip/issues/7699

Reply:
The corresponding issue now has an approved PR (#7929); so I think it'd be nice if we could adopt that approach -- it'd probably need that PR to merge but I don't think there's any reason it can't be merged within this week. :)